07_errors/n0npax/leveldbStore.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+var levelDBPath = "/tmp/leveldb"
+
+// NewLevelDBStorer creates new storer for leveldb
+func NewLevelDBStorer() *LevelDBStore {
+	db, err := leveldb.OpenFile(levelDBPath, nil)
+	dbErrorPanic(err)
+	return &LevelDBStore{nextID: 0, ldb: db}
+}
+
+// CreatePuppy creates puppy
+func (l *LevelDBStore) CreatePuppy(puppy *Puppy) (int, error) {
+	l.Lock()
+	defer l.Unlock()
+	id, err := l.putPuppy(l.nextID, puppy)
+	l.nextID++
+	return id, err
+}
+
+// ReadPuppy reads puppy from backend
+func (l *LevelDBStore) ReadPuppy(id int) (*Puppy, error) {
+	byteID := []byte(strconv.Itoa(id))
+	if puppyData, err := l.ldb.Get(byteID, nil); err == nil {
+		var puppy Puppy
+		err := json.Unmarshal(puppyData, &puppy)
+		if err != nil {
+			return nil, Errorf(ErrCodeInternal, "Internal error. Could not cast stored data to puppy object")
```

Comment:
nit: error messages should not start with capital letters, and "Internal error." is kind of unnecessary as that is specified by the code, so just "could no thas stored data to puppy object" is all that is needed as the error string.

Reply:
fixed